File: entity/src/main/java/com/marvelcomics/brito/entity/CharacterEntity.java
package com.marvelcomics.brito.entity;

public class CharacterEntity {

    private int id;
    private String name;
    private String description;
    private ThumbnailEntity thumbnailEntity;

    public CharacterEntity(int id, String name, String description, ThumbnailEntity thumbnailEntity) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.thumbnailEntity = thumbnailEntity;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public ThumbnailEntity getThumbnailEntity() {
        return thumbnailEntity;
    }

    public void setThumbnailEntity(ThumbnailEntity thumbnailEntity) {
        this.thumbnailEntity = thumbnailEntity;
    }

    public String getFullUrlThumbnailWithAspect(String aspect) {
        ThumbnailEntity thumb = this.getThumbnailEntity();
        if (aspect.isEmpty()) {
            return thumb.getPath() + "." + thumb.getExtension();
        } else {
            return thumb.getPath() + "/" + aspect + "." + thumb.getExtension();
        }
    }
}

File: domain/src/main/java/com/marvelcomics/brito/domain/entity/CharacterEntity.kt
package com.marvelcomics.brito.domain.entity

import java.io.Serializable

class CharacterEntity(
    val id: Int,
    val name: String?,
    val description: String?,
    private val thumbnailEntity: ThumbnailEntity?
) : Serializable {
    fun getFullUrlThumbnailWithAspect(aspect: String): String? {
        return if (aspect.isEmpty()) {
            thumbnailEntity?.path.toString() + "." + thumbnailEntity?.extension
        } else {
            thumbnailEntity?.path.toString() + "/" + aspect + "." + thumbnailEntity?.extension
        }
    }
}